k4FWCore/include/k4FWCore/FunctionalUtils.h

```diff
@@ -0,0 +1,19 @@
+#ifndef K4FWCORE_FUNCTIONALUTILS_H
+#define K4FWCORE_FUNCTIONALUTILS_H
+
+#include "GaudiAlg/GaudiAlgorithm.h"
+#include "GaudiKernel/DataObjectHandle.h"
+#include "k4FWCore/DataWrapper.h"
+
+// Base class used for the Traits template argument of the
+// Gaudi::Functional algorithms
+struct BaseClass_t {
+  template<typename T>
+  using OutputHandle = DataObjectWriteHandle<DataWrapper<T>>;
```

Comment:
Would this still make it possible to produce things that are note a `podio::CollectionBase`? E.g. producers that create (transient) values. Or would we have to do some SFINAE/concept magic here to select the correct Handle here?